Formula/e/envoy.rb

```diff
@@ -41,36 +42,44 @@ class Envoy < Formula
     depends_on "lld" => :build
   end
 
+  def llvm
+    deps.map(&:to_formula).find { |f| f.name.match?(/^llvm(@\d+)?$/) }
+  end
+
   # https://github.com/envoyproxy/envoy/tree/main/bazel#supported-compiler-versions
   # GCC/ld.gold had some issues while building envoy 1.29 so use clang/lld instead
   fails_with :gcc
 
   def install
     env_path = "#{HOMEBREW_PREFIX}/bin:/usr/bin:/bin"
+
+    # Override the default clang to use the one from llvm@18 in PATH
+    env_path = "#{llvm.opt_bin}:" + env_path
+
     args = %W[
       --compilation_mode=opt
       --curses=no
       --verbose_failures
       --action_env=PATH=#{env_path}
       --host_action_env=PATH=#{env_path}
       --define=wasm=disabled
+      --config=clang
+      --copt=-Wno-thread-safety-reference-return
+      --action_env=CC=#{llvm.opt_bin}/clang
+      --action_env=CXX=#{llvm.opt_bin}/clang++
+      --action_env=BAZEL_USE_CPP_ONLY_TOOLCHAIN=1
     ]
 
-    if OS.linux?
-      # GCC/ld.gold had some issues while building envoy so use clang/lld instead
-      args << "--config=clang"
-
-      # clang 18 introduced stricter thread safety analysis. Remove once release that supports clang 18
-      # https://github.com/envoyproxy/envoy/issues/37911
-      args << "--copt=-Wno-thread-safety-reference-return"
-
-      # Workaround to build with Clang 19 until envoy uses newer tcmalloc
-      # https://github.com/google/tcmalloc/commit/a37da0243b83bd2a7b1b53c187efd4fbf46e6e38
-      args << "--copt=-Wno-unused-but-set-variable"
-
-      # Workaround to build with Clang 19 until envoy uses newer grpc
-      # https://github.com/grpc/grpc/commit/e55f69cedd0ef7344e0bcb64b5ec9205e6aa4f04
-      args << "--copt=-Wno-missing-template-arg-list-after-template-kw"
+    if OS.mac?
+      # Make sure to use c90 for zlib
+      args << "--per_file_copt='external/.*zlib.*/.*.c@-std=c90'"
+      args << "--host_per_file_copt='external/.*zlib.*/.*.c@-std=c90'"
+      # Use llvm's static libs
+      args << "--linkopt=#{llvm.opt_lib}/libunwind.a"
+      args << "--linkopt=#{llvm.opt_lib}/c++/libc++.a"
+      args << "--linkopt=#{llvm.opt_lib}/c++/libc++abi.a"
+      args << "--dynamic_mode=off"
+      args << "--action_env=HOMEBREW_LIBRARY_PATHS=#{llvm.opt_bin}:#{llvm.opt_bin}/c++"
```

Comment:
Why? Using static libraries is not something we (want to) support in Homebrew/core because it requires rebuilding software for every security fix.